code/client/src/pages/Restore.jsx

```diff
@@ -128,4 +129,134 @@ const Restore = () => {
+    try {
+      const { wallet } = await retrieveWalletInfoFromAddress(retrievalAddress)
+      const innerCores = await api.blockchain.getInnerCores({ address: retrievalAddress })
+      setInnerCores(innerCores)
+      setName(name)
+      setAddress(retrievalAddress)
+      setWalletInfo(wallet)
+      setExpert(expert)
+      setInnerTrees(innerTrees)
+      setSection(Sections.SetupNewCode)
+    } catch (ex) {
+      console.error(ex)
+      message.error('Please reselect recovery file and try again.')
+    }
+  }
+
+  return (
+    <>
+      {(() => {
+        switch (section) {
+          case Sections.Choose:
+            return (
+              <AnimatedSection>
+                <Space direction='vertical' size='large' style={{ width: '100%' }}>
+                  <Heading>Scan authenticator seed QR code</Heading>
+                  <Button shape='round' size='large' type='primary' onClick={() => setSection(Sections.ScanQR)} icon={<ScanOutlined />}>Scan Now</Button>
+                  <Hint>Classic restore method. Requires webcam. Scan seed QR code exported from your authenticator (Google and Aegis Authenticator only)</Hint>
+                </Space>
+                <Divider><Hint>Or</Hint></Divider>
+                <Space direction='vertical' size='large' style={{ width: '100%' }}>
+                  <Heading>Use auth codes + recovery file</Heading>
+                  <Button shape='round' size='large' type='primary' onClick={() => setSection(Sections.SyncRecoveryFile)} icon={<FieldBinaryOutlined />}>Begin</Button>
+                  <Hint>New method. Works with all authenticators. You need (1) wallet recovery file (2) your authenticator. Provide 6-digit auth code for 6 times (30 seconds each time). Setup a new authenticator code after that.</Hint>
+                </Space>
+                <Divider><Hint>Or</Hint></Divider>
+                <Space direction='vertical' size='large'>
+                  <Heading>Sync 1wallet across devices</Heading>
+                  <LocalImport />
+                  <Hint>Requires the same 1wallet from another device. Does not require authenticator. To use this, you need to export .1wallet file under "About" tab from another device. For best security, you should delete the .1wallet file after that.</Hint>
+                </Space>
+              </AnimatedSection>
+            )
+          case Sections.ScanQR:
+            return (
+              <AnimatedSection>
+                <RestoreByScan isActive={section === Sections.ScanQR} onCancel={() => setSection(Sections.Choose)} />
+              </AnimatedSection>
+            )
+          case Sections.SyncRecoveryFile:
+            return (
+              <AnimatedSection>
+                <SyncRecoveryFile
+                  onSynced={onSynced}
+                  onCancel={() => setSection(Sections.Choose)}
+                />
+              </AnimatedSection>
+            )
+          case Sections.SetupNewCode:
```

Comment:
Need this component to render for `section === Sections.SetupNewCode || section === Sections.RecoveryCode` because it computes something in the background (then triggers a callback function to sync the result) while the user is on the next page (Sections.RecoverCode)

Reply:
the switch case can be handy sometimes, but may not be the best choice for this, personally I would avoid too many indentation levels, and sometimes does not work that great like the case pointed above.

I would say extract the sections below into separate components and then just use either `if`s or merge them into one return or do both.